File: backend/src/main/java/tp/messages/MessageType.java
package tp.messages;

import lombok.Getter;

@Getter
public enum MessageType {
    REGISTER("REGISTER"),
    LOBBY_STATUS("LOBBY_STATUS"),
    GAME_START("GAME_START"),
    GAME_LEAVE("GAME_LEAVE"),
    GAME_TRY_MOVE("GAME_TRY_MOVE"),
    GAME_MOVE("GAME_MOVE"),
    GAME_FINISHED("GAME_FINISHED"),
    REPLAY_LIST("REPLAY_LIST"),
    REPLAY_GET("REPLAY_GET"),
    LOBBY_LIST("LOBBY_LIST");

    private final String msg;

    MessageType(String msg) {
        this.msg = msg;
    }
}

File: client/Assets/ServerConnection/Messages/MessageType.cs
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace ServerConnection.Messages
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum MessageType
    {
        [EnumMember(Value = "REGISTER")]
        Register,
        [EnumMember(Value = "LOBBY_LIST")]
        LobbyList,
        [EnumMember(Value = "LOBBY_STATUS")]
        LobbyStatus,
        [EnumMember(Value = "GAME_START")]
        GameStart,
        [EnumMember(Value = "GAME_LEAVE")]
        GameLeave,
        [EnumMember(Value = "GAME_TRY_MOVE")]
        GameTryMove,
        [EnumMember(Value = "GAME_MOVE")]
        GameMove,
        [EnumMember(Value = "GAME_FINISHED")]
        GameFinished,
        [EnumMember(Value = "REPLAY_LIST")]
        ReplayList,
        [EnumMember(Value = "REPLAY_GET")]
        ReplayGet,
    }
}
